src/ExportHandlers.mss

```diff
@@ -1,52 +1,128 @@
-            // We have a template. Use the default handler.
-            handler = CreateDictionary('template', handlerDefinition);
-            if (handlerDefinition._property:createModifier)
-            {
-                handler.SetMethod('HandleObject', Self, 'ModifierTemplateHandler');
-            }
-            else
-            {
-                handler.SetMethod('HandleObject', Self, 'ControlEventTemplateHandler');
-            }
+            // Extension handlers get the API object as first argument,
+            // basically the `this` argument.
+            handler = CreateApiObject(pluginInfo._extensionInfo);
         }
-        handlers[id] = handler;
+        handler['template'] = templatesById[id];
+        handler.SetMethod('HandleObject', pluginThatDefinesHandler, handlerMethod);
+        handlers.@idProperty[id] = handler;
     }
 }   //$end
 
 
+function FindPluginThatDefinesHandler (pluginInfo, handlerMethod) {
+    // Check whether the handler method is defined in either the extension or
+    // Sibmei itself. Returns the plugin where the method was found. If a method
+    // of that name is found in both, the extension is returned. If the method
+    // was not found at all, aborts with an error.
+
+    if (pluginInfo != Self and pluginInfo._extensionInfo.plugin.MethodExists(handlerMethod))
+    {
+        // It's a bit confusing. We have two classes of objects that represent
+        // a Sibelius plugin. The one that can be referenced by Self or a plugin
+        // name (like `libmei4` if libmei4.plg is the plugin file) is needed for
+        // `SetMethod()`, the other one (that is listed in Siblius.Plugins) is
+        // needed for MethodExists().
+        // We can retrieve the former kind of plugin by its name:
+        extensionPlgName = pluginInfo._extensionInfo.plgName;
+        return @extensionPlgName;
+    }
+
+    if (SibmeiPlugin.MethodExists(handlerMethod))
+    {
+        return Self;
+    }
+
+    message = '`' & handlerMethod & '()` is not defined.';
+
+    // We didn't find the method
+    if (pluginInfo = Self)
+    {
+        StopPlugin('Internal Sibmei error: ' & message);
+    }
+
+    StopPlugin('Error in extension plugin:\n\n  ' & pluginInfo._extensionInfo.plugin.Name & '\n\n' & message);
+}  //$end
+
+
+function AssertIdType (pluginInfo, handlers, idProperty, functionName) {
+    if (null = handlers[idProperty])
+    {
+        validIdProperties = CreateSparseArray();
+        for each Name validIdProperty in handlers
+        {
+            validIdProperties.Push(validIdProperty);
+        }
+        Sibelius.MessageBox(
+            'Error in extension plugin \''
+            & pluginInfo._extensionInfo.plugin.Name
+            & '\': Expected either of \''
+            & validIdProperties.Join('\' or \'')
+            & ' as first paramter of registration function, but found \''
+            & idProperty
+            & '\'.\n\nPlugin execution is aborted. To continue, deactivate or fix \''
```

Comment:
```suggestion
            & '\'.\n\nPlugin execution is stopped. To continue, deactivate or fix \''
```

Reply:
Many thanks for looking at this so promptly, Andrew. I think I'll wait if there will be more suggestions like this and create a big commit with cosmetic/wording changes.